DOCS-4845 ASM First Steps guide (#17036)
DataDog/documentation: content/en/getting_started/application_security/_index.md

kind: documentation
further_reading:
- link: "/security/application_security/terms"
  tag: "Documentation"
  text: "Application Security terms and concepts"
- link: "/security/application_security/how-appsec-works"
  tag: "Documentation"
  text: "How Application Security Management works"
- link: "/security/application_security/getting_started"
  tag: "Documentation"
  text: "Setting up ASM"
- link: "https://securitylabs.datadoghq.com/"
  tag: "Security Labs"
  text: "Security research, reports, tips, and videos from Datadog"
---

## Overview

Datadog Application Security Management (ASM) provides observability into application-level attacks that aim to exploit code-level vulnerabilities, and into any bad actors targeting your systems. ASM helps you quickly respond to threats and remediate vulnerabilities on your web applications and APIs.

This guide walks you through best practices for getting your team up and running with ASM.

## Phase 1: Enable ASM on your exposed services

In your first three days with Datadog ASM:

1. **Identify the most exposed services.** Go to the [Service Catalog Security view][1] and sort the data by the `Suspicious Requests` column.  

>    {{< img src="getting_started/appsec/gs-appsec-most-exposed.png" alt="Service Catalog Security view sorted by the Suspicious requests column. The web-store service, for example, shows thousands of suspicious requests detected." style="width:100%;" >}}

Reviewer: Screenshot added: https://docs.google.com/document/d/1Y5UffEqmn9M2fFDmRX0P7ISVIT_XrhxzVeXMO7nGuqc/edit

>    This data is collected from APM traces (through Datadog libraries). It provides you with visibility on the services that are seeing the most suspicious traffic.

Reviewer: ```suggestion
   This data is collected from APM traces (through Datadog libraries). It provides you with visibility on the services that are exposed to the most suspicious traffic.
```

   You don't need to examine every suspicious request individually. ASM generates _signals_ when your attention is required, which you can learn about in Phase 2.

4. Now that ASM is set up detecting suspicious requests on your service in real-time, **let the product run for a couple days**.

<div class="alert alert-info">If your services use recent versions of tracing libraries, ASM immediately <a href="/security/application_security/risk_management">detects security vulnerabilities</a> in the upstream libraries that your services have as dependencies, and APM previews this information for you in its Security view. Optionally, <strong>explore ASM's consolidated <a href="https://app.datadoghq.com/security/appsec/vm">Vulnerabilities view</a></strong> (beta) to see how it helps you triage and prioritize vulnerabilities.</div>

**Note**: ASM also enables you to **track user activity**, identifying authenticated users making suspicious requests. [Setting up authenticated user tracking][5] is not required to use ASM, but it does help you get visibility on authenticated attacks and attackers. 

## Phase 2: Review your first security signals and vulnerabilities

> After a few days of usage, you typically get your first security signals. If you have not received your first security signal, you can simulate an attack and trigger a signal by running the following script, customized to hit your application's endpoints:

Reviewer: ```suggestion
After a few days of usage, you typically get your first security signals. If you have not received your first security signal, you can simulate an attack and trigger a signal by running the following script, from a terminal:
```

Author: ```suggestion
After a few days of usage, you typically get your first security signals. If you have not received your first security signal, you can simulate an attack and trigger a signal by running the following script from a terminal:
```

{{< code-block lang="sh" >}}
for ((i=1;i<=200;i++)); do
# Target existing service’s routes
curl https://your-application-url/<EXISTING ROUTE> -A
'dd-test-scanner-log';
# Target non existing service’s routes
curl https://your-application-url/<NON-EXISTING ROUTE> -A
'dd-test-scanner-log';
done{{< /code-block >}}

Over the next few days:

1. **Review security signals at [Security --> Application Security --> Signals][6].** Security signals are generated by ASM when your attention is required. The signal severity provides you with insights on how quickly you need to react. 
   
   Aim to review `INFO` and `LOW` signals once a week, `MEDIUM` and `HIGH` once a day, and `CRITICAL` signals immediately.

2. **Click a signal to see its details.** The signal details tell you what happened, who the attackers are, and what you should do next. On the Traces tab, you can investigate the traces that generated the signals.

> 3. **Archive the signal after it's been investigated.** If the signal is a false positive, set up an [exclusion filter][9] to eliminate noisy signal patterns.

Reviewer: ```suggestion
3. **Archive the signal after it's been investigated.** If the signal is a false positive, set up an [[passlist entry](https://docs.datadoghq.com/security/application_security/threats/setup_and_configure/#exclude-specific-values-from-triggering-detections)][9] to eliminate noisy signal patterns.
```

Author: ```suggestion
3. **Archive the signal after it's been investigated.** If the signal is a false positive, set up a [passlist entry][9] to eliminate noisy signal patterns.
```

> 4. **Set up a [notification][12]** to alert you in real-time when attacks require your attention, over email, Slack, or other communication integration. We recommend setting a [notification rule][11] for `Medium` and higher severity signals.

Reviewer: ```suggestion
4. **Subscribe to the [weekly digest](https://app.datadoghq.com/security/appsec/reports-configuration) to receive weekly updates on your detected security activity and set up a [notification rule](https://app.datadoghq.com/security/configuration/notification-rules) to be alerted in real-time (email, slack…) when attacks require your attention. We recommend setting a notification rule for Medium+ severity signals..
```

Author: I moved the 'subscribe' step to come before the 'notification rule' step, but they each deserve to be separate steps. Adjusted the wording a bit to reflect correct terminology, but also to follow docs style.

> 5. **Set up a [notification rule][12]** to alert you in real-time when attacks require your attention, over email, Slack, or other communication integration. Datadog recommends [setting a notification rule][11] for `Medium` and higher severity signals.

Reviewer 2: ```suggestion
5. **Set up a [notification rule][12]** to alert you in real-time when attacks require your attention, over email, Slack, or other communication integrations. Datadog recommends [setting a notification rule][11] for `Medium` and higher severity signals.
```

> 5. **[Subscribe to the weekly digest][10]** to receive weekly reports on your security signals.

Reviewer: Delete; included in 4

Author: These are very different activities from a customer perspective, and so deserve separate steps in this checklist.

> ## Further reading

Reviewer: Include https://securitylabs.datadoghq.com/

Author: it is (further reading links appear at the top of the file), see line 14
